### PR TITLE
ci(codeql): add master push hook

### PR DESCRIPTION
**Pull request by sjohnson-sage**
### Proposed behaviour
The CodeQL scanning requires scans of the base branch to compare pull requests against, so we need to add a push hook for the master branch to this workflow.

### Current behaviour
This workflow currently only runs for PRs.

image

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: CodeQL
 
-on: pull_request
+on:
+  pull_request:
```
Comment (DlgSHi) [inline .github/workflows/codeql-analysis.yml:4]: I think we need to remove this line

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (sjohnson-sage) [reply .github/workflows/codeql-analysis.yml:4]: Why? We still want this workflow to run on all PRs as well as pushes to master

Comment (DlgSHi) [reply .github/workflows/codeql-analysis.yml:4]: I think we need to delete `branch: master` and leave just `on: push`

Comment (sjohnson-sage) [reply .github/workflows/codeql-analysis.yml:4]: Oh right, I thought you were referring to the `pull_request` line!

Comment (sjohnson-sage) [reply .github/workflows/codeql-analysis.yml:4]: We've talked about this elsewhere and decided to leave it like it is for now

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Pull request merged**